src/modes/battle/battle_finish.cpp

```diff
@@ -396,16 +396,23 @@ FinishVictoryAssistant::~FinishVictoryAssistant()
     _header_window.Destroy();
     _spoils_window.Destroy();
 
-    for(uint32_t i = 0; i < _number_character_windows_created; ++i) {
+    for (uint32_t i = 0; i < _number_character_windows_created; ++i) {
         _character_window[i].Destroy();
     }
 
-    // Add all the objects that were dropped by enemies to the party's inventory
-    for(std::map<GlobalObject *, int32_t>::iterator i = _objects_dropped.begin(); i != _objects_dropped.end(); ++i) {
+    // Add all the objects that were dropped by enemies to the party's inventory.
+    for (std::map<GlobalObject*, int32_t>::iterator i = _objects_dropped.begin(); i != _objects_dropped.end(); ++i) {
+
+        // Add the item to the inventory.
+        assert(i->first != nullptr);
         GlobalManager->AddToInventory(i->first->GetID(), i->second);
```

Comment:
However, this "AddToInventory" allocates a new global object internally when needed and increments the count of a current global object when not needed.  It does nothing to the current global object pointer.